src/sys/wait.rs

```diff
@@ -267,13 +253,14 @@ impl From<i32> for PidGroup {
     }
 }
 
-impl Into<i32> for PidGroup {
-    fn into(self) -> i32 {
+impl Into<pid_t> for PidGroup {
```

Comment:
This is a public implementation that I don't think we want to support. While the first argument of `waitpid` is a `pid_t` type, it's actually not a "proper" one. As in it's different than the `pid_t` you'd get or pass to many other functions, so it's in essence a different type (it's just overloaded in C because of language limitations). I think instead we want to just hardcode this inside of `waitpid` or implement it as a private function (so it can be unit tested) instead.